File: lib/models/vendor_model.dart
class Vendor {
  final String id;
  final String companyName;
  final String? email;
  final String? location;
  final String? address;
  final String? contactPerson;
  final String? contactNumber;
  final String? profileName;
  final List<String> categories;
  final List<String> subCategories;
  final List<String> projects;
  final String? suggestedBy;
  final String? websiteUrl;
  final String? linkedinUrl;
  final String? profileDocType;
  final String? profileDocValue;
  final DateTime? createdAt;
  final DateTime? updatedAt;

  Vendor({
    required this.id,
    required this.companyName,
    this.email,
    this.location,
    this.address,
    this.contactPerson,
    this.contactNumber,
    this.profileName,
    this.categories = const [],
    this.subCategories = const [],
    this.projects = const [],
    this.suggestedBy,
    this.websiteUrl,
    this.linkedinUrl,
    this.profileDocType,
    this.profileDocValue,
    this.createdAt,
    this.updatedAt,
  });

  factory Vendor.fromJson(Map<String, dynamic> json) {
    List<String> parseList(dynamic raw) {
      if (raw == null) return [];
      if (raw is List) return raw.map((e) => e.toString()).toList();
      return [];
    }

    return Vendor(
      id: json['id']?.toString() ?? '',
      companyName: json['company_name'] ?? '',
      email: json['email'],
      location: json['location'],
      address: json['address'],
      contactPerson: json['contact_person'],
      contactNumber: json['contact_number'],
      profileName: json['profile_name'],
      categories: parseList(json['categories']),
      subCategories: parseList(json['sub_categories']),
      projects: parseList(json['projects']),
      suggestedBy: json['suggested_by'],
      websiteUrl: json['website_url'],
      linkedinUrl: json['linkedin_url'],
      profileDocType: json['profile_doc_type'],
      profileDocValue: json['profile_doc_value'],
      createdAt: json['created_at'] != null ? DateTime.tryParse(json['created_at'].toString()) : null,
      updatedAt: json['updated_at'] != null ? DateTime.tryParse(json['updated_at'].toString()) : null,
    );
  }
}

File: f_b/backend/src/models/vendor.model.js
const sheets = require("../config/googleSheet");
require("dotenv").config();

const SPREADSHEET_ID = process.env.SPREADSHEET_ID;
const SHEET_NAME = "vendors";

const createVendorSheet = async () => {
  try {
    // ✅ Create Sheet if Missing
    try {
      await sheets.spreadsheets.batchUpdate({
        spreadsheetId: SPREADSHEET_ID,
        requestBody: {
          requests: [
            {
              addSheet: {
                properties: { title: SHEET_NAME },
              },
            },
          ],
        },
      });
      console.log("✅ Vendors sheet created");
    } catch {
      console.log("⚡ Vendors sheet already exists");
    }

    // ✅ Header Row
    await sheets.spreadsheets.values.update({
      spreadsheetId: SPREADSHEET_ID,
      range: `${SHEET_NAME}!A1:T1`,
      valueInputOption: "RAW",
      requestBody: {
        values: [
            [
 "id",
 "vendor_id",
 "company_name",
 "email",
 "location",
 "address",
 "contact_person",
 "contact_number",
 "profile_name",

 "categories",
 "sub_categories",

 "suggested_by",
 "website_url",
 "linkedin_url",
 "projects",

 "profile_doc_type",
 "profile_doc_value",

 "created_at",
 "updated_at"
]

    ],
      },
    });

    console.log("✅ Vendor Sheet Ready");
  } catch (err) {
    console.error("❌ Vendor Sheet Error:", err);
    throw err;
  }
};

module.exports = { createVendorSheet };
